src/components/Api.js

```diff
@@ -0,0 +1,84 @@
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkServerResponse(res) {
```

Comment:
It's great to have a method for processing the server response, so the logic is not duplicated